mobtravelapp/src/main/java/com/mobiquity/mobtravelapp/service/CalendarService.java

```
@@ -0,0 +1,88 @@
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.DateTime;
+import com.google.api.client.util.store.FileDataStoreFactory;
+import com.google.api.services.calendar.Calendar;
+import com.google.api.services.calendar.CalendarScopes;
+import com.google.api.services.calendar.model.Event;
+import com.google.api.services.calendar.model.Events;
+import org.springframework.stereotype.Service;
+
+import java.io.InputStreamReader;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+
+@Service
+public class CalendarService {
+
+    private static HttpTransport httpTransport;
+
+    /** Directory to store user credentials. */
+    private static final java.io.File DATA_STORE_DIR =
+            new java.io.File(System.getProperty("user.home"), ".store/mob_calendar_app");
+
+    private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+
+    private static FileDataStoreFactory dataStoreFactory;
+
+    public List<Event> getEvents(){
+
+        List<Event> eventsList = new ArrayList<>();
+        try {
+            httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+            dataStoreFactory = new FileDataStoreFactory(DATA_STORE_DIR);
+
+            Calendar service = new Calendar.Builder(httpTransport, JSON_FACTORY, authorize())
+                    .setApplicationName("MobCalendarApp").build();
+
+            String pageToken = null;
+            do {
+                Events events = service.events().list("primary")
+                        .setTimeMin(new DateTime(Date.from(Instant.now())))
+                        .setPageToken(pageToken).execute();
+                List<Event> items = events.getItems();
+                pageToken = events.getNextPageToken();
+                eventsList = items;
+            } while (pageToken != null);
+
+        } catch (Exception e) {
+            e.printStackTrace();
```

Swallowing exceptions is really BAD

---

printing to the console almost as bad